components/app-core/backend/auth.go

```diff
@@ -154,17 +172,52 @@ func (p *portalProxy) loginToCNSI(c echo.Context) error {
 }
 
 func (p *portalProxy) DoLoginToCNSI(c echo.Context, cnsiGUID string) (*interfaces.LoginRes, error) {
-	uaaRes, u, cnsiRecord, err := p.fetchToken(cnsiGUID, c)
-
-	if err != nil {
-		return nil, err
-	}
-
 	// save the CNSI token against the Console user guid, not the CNSI user guid so that we can look it up easily
 	userID, err := p.GetSessionStringValue(c, "user_id")
 	if err != nil {
 		return nil, echo.NewHTTPError(http.StatusUnauthorized, "Could not find correct session value")
 	}
+
```

Comment:
I think this should be done via a login hook which executes when the user logs into Console.

As is implemented here, anytime the user connects to an endpoint, it will first try and re-use the console token. The user might be using a different set of credentials, which would be ignored and the user would just connect as normal with the console token.

Search for LoginHook in the code for how this is done when deployed in Cloud Foundry.

Reply:
@nwmac ok good point. can you look at the front end tests? there is just one that's failing and I think it has to do with your changes since I didn't change front end.